src/model/cloud-runner/cloud-runner.ts

```diff
@@ -1,69 +1,88 @@
+import CloudRunnerEnvironmentVariable from './services/cloud-runner-environment-variable';
+import TestCloudRunner from './cloud-runner-providers/test';
+import LocalCloudRunner from './cloud-runner-providers/local';
+import LocalDockerCloudRunner from './cloud-runner-providers/local-docker';
 
 class CloudRunner {
+  public static CloudRunnerProviderPlatform: CloudRunnerProviderInterface;
+  static buildParameters: BuildParameters;
+  public static defaultSecrets: CloudRunnerSecret[];
+  public static cloudRunnerEnvironmentVariables: CloudRunnerEnvironmentVariable[];
   private static setup(buildParameters: BuildParameters) {
     CloudRunnerLogger.setup();
-    CloudRunnerState.setup(buildParameters);
+    CloudRunner.buildParameters = buildParameters;
     CloudRunner.setupBuildPlatform();
-    const parameters = TaskParameterSerializer.readBuildEnvironmentVariables();
-    for (const element of parameters) {
-      core.setOutput(element.name, element.value);
+    CloudRunner.defaultSecrets = TaskParameterSerializer.readDefaultSecrets();
+    CloudRunner.cloudRunnerEnvironmentVariables = TaskParameterSerializer.readBuildEnvironmentVariables();
+    if (!buildParameters.cliMode) {
+      const buildParameterPropertyNames = Object.getOwnPropertyNames(buildParameters);
+      for (const element of CloudRunner.cloudRunnerEnvironmentVariables) {
+        core.setOutput(Input.ToEnvVarFormat(element.name), element.value);
+      }
+      for (const element of buildParameterPropertyNames) {
+        core.setOutput(Input.ToEnvVarFormat(element), buildParameters[element]);
+      }
     }
   }
 
   private static setupBuildPlatform() {
-    switch (CloudRunnerState.buildParams.cloudRunnerCluster) {
+    CloudRunnerLogger.log(`Cloud Runner platform selected ${CloudRunner.buildParameters.cloudRunnerCluster}`);
+    switch (CloudRunner.buildParameters.cloudRunnerCluster) {
       case 'k8s':
-        CloudRunnerLogger.log('Cloud Runner platform selected Kubernetes');
-        CloudRunnerState.CloudRunnerProviderPlatform = new Kubernetes(CloudRunnerState.buildParams);
+        CloudRunner.CloudRunnerProviderPlatform = new Kubernetes(CloudRunner.buildParameters);
         break;
-      default:
       case 'aws':
-        CloudRunnerLogger.log('Cloud Runner platform selected AWS');
-        CloudRunnerState.CloudRunnerProviderPlatform = new AWSBuildPlatform(CloudRunnerState.buildParams);
+        CloudRunner.CloudRunnerProviderPlatform = new AWSBuildPlatform(CloudRunner.buildParameters);
+        break;
+      case 'test':
+        CloudRunner.CloudRunnerProviderPlatform = new TestCloudRunner();
+        break;
+      case 'local-system':
+        CloudRunner.CloudRunnerProviderPlatform = new LocalCloudRunner();
+        break;
+      case 'local-docker':
+        CloudRunner.CloudRunnerProviderPlatform = new LocalDockerCloudRunner();
         break;
     }
   }
```

Comment:
I love that you used a strategy pattern for this. It'll come in handy soon enough with our current plans.